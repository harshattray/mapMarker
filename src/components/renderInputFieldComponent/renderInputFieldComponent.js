/**
 * @Author: harsha
 * @Date:   2019-04-18T00:54:07+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-25T03:39:27+05:30
 */
import React, { Fragment } from "react";
import { Input } from "semantic-ui-react";

/**
 * [RenderInputFields description]
 * @param  {[type]} label       [Conditionally renders label based on props]
 * @param  {[type]} type        [Type of form to be rendered]
 * @param  {[type]} placeholder [Conditionally rendered placeholder based on props]
 * @param  {[type]} name        [description]
 * @param  {[type]} pattern     [description]
 * @param  {[type]} meta        [description]
 * @param  {[type]} error       [description]
 * @param  {[type]} warning     [description]
 * @param  {[type]} input       [Input object]
 * @return {[type]}             [description]
 */

export const RenderInputFields = ({
  label,
  type,
  placeholder,
  name,
  pattern,
  mapError,
  meta: { touched, error, warning },
  input
}) => {
  return (
    <Fragment>
      <Input type={type} label={label} placeholder={placeholder} {...input} />
      <div>{touched && error && <i>{error}</i>}</div>
      <div>{mapError && <i>{mapError}</i>}</div>
    </Fragment>
  );
};
