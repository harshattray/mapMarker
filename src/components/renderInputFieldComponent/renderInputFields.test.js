/**
 * @Author: harsha
 * @Date:   2019-04-24T23:54:01+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-24T23:59:20+05:30
 */

import React from "react";
import { cleanup, render } from "react-testing-library";

import { RenderInputFields } from "./renderInputFieldComponent";

describe("Component => RenderInputFields", () => {
  afterEach(cleanup);
  it("should match snapshot", () => {
    const props = {
      input: { name: "MOCK_NAME" },
      label: "MOCK_LABEL",
      type: "text",
      meta: {
        touched: true,
        error: false,
        warning: false
      }
    };

    const { container } = render(<RenderInputFields {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
