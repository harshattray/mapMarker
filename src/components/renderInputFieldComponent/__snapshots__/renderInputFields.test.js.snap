// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component => RenderInputFields should match snapshot 1`] = `
<div
  class="ui labeled input"
>
  <div
    class="ui label label"
  >
    MOCK_LABEL
  </div>
  <input
    name="MOCK_NAME"
    type="text"
  />
</div>
`;
